### PR TITLE
Implemented missing special enums.

### LINKED ISSUE #9: Implement missing special enums
### Caution
These might not be in the locations they're supposed to be. It might also not be the only location they are (supposed to be) in. This can be the case for all charts we've not reworked yet.  
Also the list might be incomplete.

- [ ] `FontStyle` in Labels.cs
- [ ] `BorderCapStyle` in LineDataset.cs
- [ ] `BorderJoinStyle` in LineDataset.cs
- [ ] `CubicInterpolationMode` in LineDataset.cs
- [ ] `PointStyle` in BubbleDataset.cs
- [ ] `BorderSkipped` in DoughnutDataset.cs
- [ ] `BorderAlign` in PieDataset.cs

### PR DESCRIPTION
**Pull request by SeppPenner**
Fixes https://github.com/Joelius300/ChartJSBlazor/issues/9.

### CODE REVIEW

**Review by Joelius300 — COMMENTED** [review]
I think both commits for the project-file ([this](https://github.com/Joelius300/ChartJSBlazor/pull/11/commits/faf31e011df0478174399e105469f784b23499c0) and [this](https://github.com/Joelius300/ChartJSBlazor/pull/11/commits/067878699236f4eccdd8e5409cc53f0692d55697)) are good. Maybe we shouldn't say "See the samples" since they're not really existent yet :).  

Regarding the enum-implementations however I have a few issues. First of all the current way these are implemented is with the StringEnum class. Furthermore they are almost all located in the namespace `..Common.Enums`. Because there are some enums that have to support multiple types (e.g. [SteppedLine](https://github.com/Joelius300/ChartJSBlazor/blob/master/ChartJs.Blazor/ChartJS/Common/Enums/SteppedLine.cs)) we definitely cannot get rid of `ObjectEnum`. However `StringEnum` could be removed since we can just use `[Newtonsoft.Json.JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]`. However as you can see this still uses json.net and I have no idea how you'd do it with `System.Text.Json` - even though there is a `JsonStringEnumConverter` you cannot use the `JsonConverterAttribute` on an enum.. maybe that's a mistake or maybe you don't use it in this way. See [this github issue](https://github.com/dotnet/corefx/issues/39741) for updates on that.

Either we use normal enums for all of these and rework the old ones or we rework the new ones you added to use `StringEnum` as well.  
If we keep the `StringEnum`s we don't have to worry about capitalization etc and can name the Properties however we want which is slightly more pretty.  
If we use real `enum`s for those with the correct attribute, we would have constant values and an easier implementation. Another advantage would be that it might be easier to port to `System.Text.Json` but that's just speculation.  

I'm glad for some more thoughts about this issue so we can decide on a path.

**Comment by SeppPenner** [comment]
Well, I would use real enums and try to tell the `JsonConverter` somehow to capitalize them correctly (No idea how this works in `System.Text.Json` though). In general, I don't really like to have these `StringEnum` or `ObjectEnum`... I suppose there must be a way to do this much easier. I will check this again and see what I can find.

**Comment by Joelius300** [comment]
It's possible to get rid of `StringEnum` and just use `enum`. However I'm almost certain that there is no way to get rid of `ObjectEnum`. For example `SteppedLine` has both a value `false` and a value `"before"`. If we only use an `enum` (or `StringEnum` - doesn't matter), it will parse `SteppedLine.False` to `"false"` which will not be interpreted by js as a `boolean` value and therefore break Chart.Js (I have tested that).

**Comment by SeppPenner** [comment]
> It's possible to get rid of `StringEnum` and just use `enum`. However I'm almost certain that there is no way to get rid of `ObjectEnum`. For example `SteppedLine` has both a value `false` and a value `"before"`. If we only use an `enum` (or `StringEnum` - doesn't matter), it will parse `SteppedLine.False` to `"false"` which will not be interpreted by js as a `boolean` value and therefore break Chart.Js (I have tested that).

How did you even manage to serialize enums to strings without writing a custom formatter? I didn't see that this is even supported yet in `System.Text.Json`?

**Comment by Joelius300** [comment]
I don't know if you've read through all the stuff at https://github.com/Joelius300/ChartJSBlazor/issues/2 (I know there's _a lot_ of stuff) but there you can find the answer to your question.  

### tl;dr

We need to get rid of `null` values (not serialize them) because they break Chart.Js.  
Since `IJSRuntime` (which is used for the js-interop) doesn't support using your own serializer-options, we have to prepare the data first. This is done via json.net and an `ExpandoObject`. It's really ugly and bad but for one, we can use json.net custom converters and also we can prepare the instance so it doesn't have null values anymore.  


We can in fact fix this and I will maybe add another issue for _Migration from json.net to System.Text.Json_.  
The custom converters can be rewritten to get the same functionality as those we use currently. It is supported, however not as well as with json.net (it doesn't work if you just put it on base-classes currently).  
They also have `StringEnumConverter` which had an issue that you couldn't place it on enums. This is fixed in the master branch but not publicly available through a preview.  

There are more things and issues which prevent me from fully migrating just yet. Please take a look at [my help repos](https://github.com/Joelius300/serialization-tests) where you can see how things are serialized. I have also done a lot of testing around this kind of stuff and you can see in that project that I have found workarounds for many of the things which prevent us from switching. Sadly not all of them.  
Also there is [this tracking issue](https://github.com/Joelius300/serialization-tests/issues/1) where you can look at what I've done over at `corefx` and `aspnetcore` if you're interested.

**Comment by SeppPenner** [comment]
> We need to get rid of null values (not serialize them) because they break Chart.Js.

Yep, I have had this error as well. But Text.Json can do this: https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.ignorenullvalues?view=netcore-3.0#System_Text_Json_JsonSerializerOptions_IgnoreNullValues

> They also have StringEnumConverter which had an issue that you couldn't place it on enums. This is fixed in the master branch but not publicly available through a preview.

Yeah, it's not available yet in the nuget package. Haven't seen it yet.

>There are more things and issues which prevent me from fully migrating just yet. Please take a look at my help repos where you can see how things are serialized. I have also done a lot of testing around this kind of stuff and you can see in that project that I have found workarounds for many of the things which prevent us from switching. Sadly not all of them.
Also there is this tracking issue where you can look at what I've done over at corefx and aspnetcore if you're interested.

I will check this, thank you ;)
